pkg/kubelet/secret/secret_manager.go

```diff
@@ -201,8 +201,12 @@ type cachingSecretManager struct {
 }
 
 func NewCachingSecretManager(kubeClient clientset.Interface) (Manager, error) {
+	// TODO: Currently apiserver completely cannot keep up with 1 minute ttl in
+	// 5000-node clusters. So before we optimize apiserver and/or etcd, increase
+	// the TTL to 5 minutes (which is a lot, but...)
+	ttl := 5 * time.Minute
```

Comment:
+1. This seems a bit excessive. IIUC, in kubemark, all (pause) pods on the node share the same secret. This results in 5000 requests per minute for fetching secrets in the steady state (excl. the pod startup phase). If even this is too much, a less-trivial workload with N (N>5) unique secrets per node will easily overwhelm the apiserver. Another thing is that no one runs 5000-node clusters in production right now. Optimizing the TTL for such scale while affecting the latency for the majority users doesn't seem like a good trade-off. Does this just mean we have to prioritize finding ways to watch a limited set of secrets? (or perhaps we can let apiserver return a TTL for the secret based on its load)?